### ISSUE TITLE
ROC and AUPR

### ISSUE THREAD

**Issue by Z-X20A** [lingchen0331/UQ_ICL#1]
How to calculate ROC and AUPR of EU or AU? I did not find the related code in repo. Can you provide the related code?

**Comment by lingchen0331** [comment]
Hi, there's no specific code for calculating ROC and AUPR, you may directly use [`sklearn.metrics.roc_auc_score`](https://scikit-learn.org/stable/modules/generated/sklearn.metrics.roc_auc_score.html) and ['sklearn.metrics.average_precision_score'](https://scikit-learn.org/stable/modules/generated/sklearn.metrics.average_precision_score.html#sklearn.metrics.average_precision_score) to make the calculation by using the misclassification label as the label and the AU/EU as the weight.

**Comment by Z-X20A** [comment]
Hi，lingchen,
I try to calculate use those two api. And I get some error what you means use EU/AU as weight? The weight is y-scores? If so, seems like we have three different labels, and for each question we own have one EU/AU value. Therefore, it reports an error. Can you provide more details about how to calculate it, I will appreciate it.
Thanks

---- Replied Message ----
| From | Chen ***@***.***> |
| Date | 04/01/2024 21:15 |
| To | lingchen0331/UQ_ICL ***@***.***> |
| Cc | Z-X20A ***@***.***>,
Author ***@***.***> |
| Subject | Re: [lingchen0331/UQ_ICL] ROC and AUPR (Issue #1) |

Hi, there's no specific code for calculating ROC and AUPR, you may directly use sklearn.metrics.roc_auc_score and 'sklearn.metrics.average_precision_score' to make the calculation by using the misclassification label as the label and the AU/EU as the weight.

**Comment by lingchen0331** [comment]
Hi, pls use the following code for the evaluation.
```
df = pd.read_json("results/70b_sentiment_processed_class.json")
df = df.replace(to_replace='None', value=np.nan).dropna()

valid_predictions = df["Preds"] != "None"

# For these rows, compute the "Is_Correct" column
df.loc[valid_predictions, "Is_Correct"] = (df.loc[valid_predictions, "Label"] != df.loc[valid_predictions, "Preds"].astype(int)).astype(int)

# Calculating the accuracy
accuracy = accuracy_score(df["Label"], df["Preds"].astype(int))

# Calculating AUPR
average_precision_AU = average_precision_score(df["Is_Correct"], df["AU_new"])
average_precision_EU = average_precision_score(df["Is_Correct"], df["EU_new"])

# Calculating AUROC
roc_auc_AU = roc_auc_score(df["Is_Correct"], df["AU_new"])
roc_auc_EU = roc_auc_score(df["Is_Correct"], df["EU_new"])
```
